aws/resource_aws_db_instance_test.go

```diff
@@ -2703,6 +2731,15 @@ func testAccCheckAWSDBInstanceNoSnapshot(s *terraform.State) error {
 	return nil
 }
 
+func testAccCheckAWSDBInstanceNotRecreated(instance1, instance2 *rds.DBInstance) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		if aws.TimeValue(instance1.InstanceCreateTime) != aws.TimeValue(instance2.InstanceCreateTime) {
```

Comment:
Reference: https://github.com/hashicorp/terraform-provider-aws/pull/17990

```suggestion
		if !aws.TimeValue(instance1.InstanceCreateTime).Equal(aws.TimeValue(instance2.InstanceCreateTime)) {
```